src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4559,9 +4526,11 @@ private object CreateInstanceDefaultCtorSlow(bool publicOnly, bool wrapException
         internal object CreateInstanceDefaultCtor(bool publicOnly, bool skipCheckThis, bool fillCache, bool wrapExceptions)
         {
             // Call the cached 
-            ActivatorCacheEntry? cacheEntry = s_ActivatorCache?.GetEntry(this);
+            ActivatorCacheEntry? cacheEntry = GenericCache as ActivatorCacheEntry;
             if (cacheEntry != null)
             {
+                cacheEntry.EnsureInitialized();
```

Comment:
I think that this double initialization could be avoided. Instead of `RuntimeTypeHandle.CreateInstance` you can try to construct a readonly cache entry. I don't have very deep knowledge about internals of Core CLR, so there are some questions. How long does a `RuntimeType` reside in memory? Will its cache be garbaged if the type is collectible?

Reply:
> I think that this double initialization could be avoided.

This double initialization was present in the existing code. I assume that it was done this way to minimize cost added to one-time Activator.CreateInstance invocation. It is likely that majority of Activator.CreateInstance calls are done exactly once for given type, and so it is desirable to delay the delegate allocation to the second call for give type. I agree that there may be opportunity to make this better, but it requires more refactoring on top what I have done. It can be done in follow up PR.

>  Will its cache be garbaged if the type is collectible?

Yes, this cache will be collected when the type is unloaded. Also, this cache will be collected when  not used for some time.

Reply:
> I agree that there may be opportunity to make this better, but it requires more refactoring on top what I have done. It can be done in follow up PR.

I would like to take it and learn more about internals.